backend/src/test/groovy/com/capstone/moneytree/controller/UserControllerTest.groovy

```diff
@@ -235,4 +240,114 @@ class UserControllerTest extends Specification {
       then: "Should throw CredentialNotFoundException"
       thrown(CredentialNotFoundException)
    }
+
+   @Test
+   def "Edit User is successful if he does not have an avatar url"() {
+      setup:
+      def imageUrl = "www.s3.bucket.image.com/id"
+      and:
+      amazonS3Service.uploadImageToS3Bucket(_, _) >> imageUrl
+
+      and: "A registered user with an Alpaca key"
+      String email = "moneytree@test.com"
+      String username = "Billy"
+      String password = "encrypted"
+      String firstName = "Billy"
+      String lastName = "Bob"
+      String alpacaApiKey = "RYFERH6ET5etETGTE6"
+      User user = createUser(email, username, password, firstName, lastName, alpacaApiKey)
+
+      and: "mock the database with the same user already registered"
+      userDao.findUserById(user.getId()) >> user
+
+      and: "A valid multipartFile is passed"
+      MultipartFile imageFile = Mock()
```

Comment:
Minor comment: should we use a real image to test here? Or is this going to be done in #165 ?

Reply:
This is not in scope for #165. #165 is about using the test db, unrelated to s3.

Reply:
I am mocking it for now because I wanted to tests the error paths.

The reason why I did not do an integration test is because of s3 limitations. I dont want to endup like we did with IEXCloud